java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiSliceScanner.java

```diff
@@ -88,68 +94,92 @@ public HudiSliceScanner(int fetchSize, Map<String, String> params) {
         this.classLoader = this.getClass().getClassLoader();
     }
 
-    @Override
-    public void open() throws IOException {
-        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
-            Properties properties = new Properties();
-            Configuration conf = new Configuration();
-            conf.setBoolean("dfs.client.use.legacy.blockreader", false);
-            JobConf jobConf = new JobConf(conf);
-            jobConf.setBoolean("hive.io.file.read.all.columns", false);
-            String[] hiveColumnNames = this.hiveColumnNames.split(",");
-            HashMap<String, Integer> hiveColumnNameToIndex = new HashMap<>();
-            HashMap<String, String> hiveColumnNameToType = new HashMap<>();
-            for (int i = 0; i < hiveColumnNames.length; i++) {
-                hiveColumnNameToIndex.put(hiveColumnNames[i], i);
-                hiveColumnNameToType.put(hiveColumnNames[i], hiveColumnTypes[i]);
-            }
+    private JobConf makeJobConf(Properties properties) {
+        Configuration conf = new Configuration();
+        conf.setBoolean("dfs.client.use.legacy.blockreader", false);
+        JobConf jobConf = new JobConf(conf);
+        jobConf.setBoolean("hive.io.file.read.all.columns", false);
```

Comment:
after set this config, hudi scanner can only read the columns that query used? or just read select fields of struct column?

Reply:
It affect top-level selected fields, but it will not affect nested fields in struct. 

Following one will affect nested fields in struct.

<img width="702" alt="image" src="https://user-images.githubusercontent.com/1081215/212542326-672fa371-9729-4dc4-bf2c-afca791ff106.png">